website/blog/2022-12-29-sqllogictest-framework.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Rewrite sqllogictest framework in rust
+description: sqllogictest, rust
+date: 2022-12-29
+tags: [databend, sqllogictest, rust]
+authors:
+- name: Xudong
+  url: https://github.com/xudong963
+  image_url: https://github.com/xudong963.png
+---
+
+Sqllogictest is a program designed to verify that an SQL database engine computes correct results by comparing the results to identical queries from other SQL database engines. Sqllogictest was originally designed to test [SQLite](http://www.sqlite.org/), but it is database engine neutral and can just as easily be used to test other database products.
+
+In the rust ecosystem, [sqllogictest-rs](https://github.com/risinglightdb/sqllogictest-rs) is a very good implementation of the sqllogictest framework, thanks to which databend can easily and quickly switch the sqllogictest framework from python to rust.
+
+### Background
+
+- For some historical reasons, databend's original sqllogictest framework was the python version, as seen in [rfc for sqllogictest](https://databend.rs/doc/contributing/rfcs/new_sql_logic_test_framework)
+- We decided to rewrite it in rust for the following reasons:
+  - Unified codebase, all with rust, long-term can improve the framework development and iteration speed, after all, the whole team is rustacean.
```

Comment:
Advice from chatgpt:

> By having a unified codebase written in Rust, we can improve the speed of development and iteration on our framework in the long term. This is because the entire team is proficient in Rust.